.circleci/config.yml

```diff
@@ -1,17 +1,175 @@
 version: 2.1
-jobs:
-  build:
+
+executors:
+  go:
     docker:
       - image: circleci/golang:1.14
+    environment:
+      CONSUL_VERSION: 1.7.2
+      GOMAXPROCS: 4
+      GO111MODULE: "on"
+      GOPROXY: https://proxy.golang.org/
+      TEST_RESULTS_DIR: &TEST_RESULTS_DIR /tmp/test-results
+      ARTIFACTS_DIR: &ARTIFACTS_DIR /tmp/artifacts
+      GOFLAGS: "-mod=vendor"
+
+jobs:
+  go-checks:
+    executor:
+      name: go
+    steps:
+      - checkout
+      - run: go mod verify
+      - run: make fmtcheck generate
+      - run:
+          name: verify no code was generated
+          command: |
+            if [[ -z $(git status --porcelain) ]]; then
+              echo "clean"
+            else
+              echo "dirty"
+              git status --porcelain
+              exit 1
+            fi
+
+  go-test:
+    executor:
+      name: go
+    parallelism: 4
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          name: install consul
+          command: |
+            curl -sLo consul.zip https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_linux_amd64.zip
+            unzip consul.zip
+            mkdir -p ~/bin
+            mv consul ~/bin
+            echo 'export PATH="~/bin:$PATH"'
+      - run: mkdir -p $TEST_RESULTS_DIR
+      - run: |
```

Comment:
I noticed this one (and its equivalent under `go-test-e2e`) generate quite a clunky heading in the CircleCI UI.

![](https://user-images.githubusercontent.com/20180/78052573-2d13fe00-7334-11ea-824e-eddb5ea984de.png)

Looking at some of the other examples in this file it looks like we can set `name` to customize caption on those. Could we do that here to set it to something shorter like "Running Tests"? :thinking:

Reply:
I added the naming per your request!